internal/MozziGuts_impl_RP2040.hpp

```diff
@@ -142,30 +143,60 @@ inline void audioOutput(const AudioOutput f) {
 #    endif
 }
 #  endif  // MOZZI_OUTPUT_PWM
-
+  */
 } // namespace MozziPrivate
 #include <pico/time.h>
 namespace MozziPrivate {
 /** Implementation notes:
- *  - For the time being this port uses a very crude approach to audio output: PWM updated by a hardware timer running at MOZZI_AUDIO_RATE
- *  - Hardware timer isn't fixed, but rather we claim the first unclaimed one
- *  - Quite pleasently, the RP2040 latches PWM duty cycle, so we do not have to worry about updating whilst in the middle of the previous PWM cycle.
- *  - The more simple add_repeating_timer_us has appers to have far too much jitter
- *  - Using DMA transfers, instead of a manual timer, would be much more elegant, but I'll leave that as an exercise to the reader ;-)
- *  - Not to mention PWM output, etc.
+ *  - For once, two different approaches are used between EXTERNAL_TIMED and PWM:
+ *  - EXTERNAL_TIMED (here), uses a repeating alarm to induce the user's callback
+ *  - because the alarm only has a resolution of 1us, we need to trick a bit to get the correct frequency (see below).
  */
 absolute_time_t next_audio_update;
 uint64_t micros_per_update;
 uint audio_update_alarm_num;
+bool flip_flop=0;
+// NOTE: unfortunately, on the RP2040, alarms can only be set with us resolution. At 32768Hz, we ideally would like to output every 30.51us, so we alternate between outputting after 30us and 31us, using the flip_flop to alternate between these two values,
 
 void audioOutputCallback(uint) {
   do {
     defaultAudioOutput();
-    next_audio_update = delayed_by_us(next_audio_update, micros_per_update);
+    flip_flop = !flip_flop;;
```

Comment:
I was rather thinking along the following lines (pseudocode):

```
uint64_t next_audio_update_shifted = 0;
const uint64_t us_persample_shifted = (1000000l << 8) / MOZZI_AUDIO_RATE;

[...]

next_audio_update_shifted += us_per_sample_shifted;
while(hardware_alarm_set_target(alarm_num, delayed_by_us(0, next_audio_update_shifted >> 8));
```

Reply:
Ah, I understand what you meant now! 

I guess it would even more simpler be: `while (hardware_alarm_set_target(audio_update_alarm_num, next_audio_update_shifted>>8)); // absolute time, so no need for delayed_by_us`

Indeed, that would be more general, I can try. My version works only for this case where we are basically nearly exactly between two values. I have a hard time trying to intuitively figure out which one is faster but yours might be (you: 1 add, 1 shift, me: 2 add, 1 shift). I'll try!

Reply:
@tfry-git What do you think of the new version? Note the needed addition to `startMozzi()`